injectproxy/routes.go

```diff
@@ -260,20 +261,36 @@ func (r *routes) enforceLabel(h http.HandlerFunc) http.Handler {
 // sent through a URL parameter.
 // It returns an error when either the value is found in both places, or is not found at all.
 func (r *routes) getLabelValue(req *http.Request) (string, error) {
-	formValue := req.FormValue(r.label)
-	if r.labelValue != "" && formValue != "" {
+	err := req.ParseForm()
+	if err != nil {
+		return "", fmt.Errorf("the query parameter can not be parsed. %s", err.Error())
+	}
+	formValues, _ := req.Form[r.label]
+
+	if r.labelValue != "" && len(formValues) != 0 {
 		return "", fmt.Errorf("a static value for the %s label has already been specified", r.label)
 	}
 
-	if r.labelValue == "" && formValue == "" {
+	if r.labelValue == "" && len(formValues) == 0 {
 		return "", fmt.Errorf("the %q query parameter must be provided", r.label)
 	}
 
+	// Check if every value is non-empty
+	for i := range formValues {
+		if formValues[i] == "" {
+			return "", fmt.Errorf("the %q query parameter must be provided", r.label)
+		}
+
+		formValues[i] = regexp.QuoteMeta(formValues[i])
+	}
+
 	if r.labelValue != "" {
 		return r.labelValue, nil
```

Comment:
I'm not sure that we should bypass regexp quoting when `-label-value` is defined.

In the current PR version, you could match on multiple static values doing:

```console
./prom-label-proxy -label project -label-value 'prometheus|alertmanager'
```

But I feel that the following would be safer and more consistent with the "label value via query parameter" mode

```console
./prom-label-proxy -label project -label-value prometheus -label-value alertmanager
```

Reply:
> I'm not sure that we should bypass regexp quoting when `-label-value` is defined.
> 
> In the current PR version, you could match on multiple static values doing:
> 
> ```
> ./prom-label-proxy -label project -label-value 'prometheus|alertmanager'
> ```
> 
> But I feel that the following would be safer and more consistent with the "label value via query parameter" mode
> 
> ```
> ./prom-label-proxy -label project -label-value prometheus -label-value alertmanager
> ```

I agree with you completely and have changed the code to the consistent way.